backend/src/users/services/profile_service.py

```diff
@@ -0,0 +1,194 @@
+            data = cache_handler.get_event()
+            if data:
+                print(data)
+                user_id = UUID(data["id"])
+                profile = data["profile"]
+                self.create_related_profile(
+                    user_id=user_id,
+                    profile=profile,
+                    db=self.db,
+                )
+                cache_handler.unsubscribe_event("user_created")
+
+    def event_handler_user_deleted(
+        self,
+        cache_handler: Optional[CacheHandler],
+        subscribe_name: Optional[str],
+    ):
+        cache_handler.subscribe_event(subscribe_name)
+        while True:
+            time.sleep(0.2)
+            data = cache_handler.get_event()
+            if data:
+                print(data)
+                user_id = UUID(data["id"])
+                self.delete_related_profile(
+                    user_id=user_id,
+                )
+                cache_handler.unsubscribe_event("user_deleted")
+
+    def create_related_profile(
+        self,
+        user_id: UUID,
+        profile: CreateProfileModel,
+        db: Session,
+    ):
+        try:
+            # Create profile in the repository
+            profile_db = self.repository.create_profile(
+                user_id=user_id,
+                name=profile.get("name"),
+                surname=profile.get("surname"),
+                db=db,
+            )
+            if profile_db:
+                # Prepare and upload profile image to cloud storage
+                image = prepare_profile_image(
+                    image_url=profile.get("image_url"),
+                    user_id=user_id,
+                )
+
+                # Upload profile image to cloud storage
+                self.cloud_storage.upload_file(
+                    **image,
+                )
+            else:
+                raise InvalidProfileData
+
+        except Exception:
+            raise ProfileCreationFailed
+
+    @staticmethod
+    def delete_related_profile(user_id: UUID):
+        try:
+            delete_profile_image(user_id=user_id)
+            return True
+        except Exception:
+            raise DeleteFailed
+
+    def get_profile(
+        self,
+        user_id: UUID,
+        db: Session,
+    ) -> Optional[JSONResponse]:
+        try:
+            profile_db = self.repository.get_profile_by_user_id(user_id=user_id, db=db)
+            if not profile_db:
+                raise InvalidProfileData
+            image_base64 = get_profile_image(
+                user_id=user_id,
+            )
+
+            if not image_base64:
+                raise HTTPException(
+                    status_code=404,
+                    detail="Profile image not found",
+                )
+            return JSONResponse(
+                content=ProfileDetailResponse(
+                    name=profile_db.name,
+                    surname=profile_db.surname,
+                    image_base64=image_base64,
```

Comment:
In cloud_storage is existing method to read images from minio. 

You shouldn't use base64 for photo images, because if we load base64 on html, this will be too heavy and leading to increased page load times and potentially degrading the user experience. Base64 is used only for small images, such as icons.